app/views/employees/_form.html.erb

```diff
@@ -1,5 +1,11 @@
 <%= simple_form_for @employee do |form| %>
-  <%= form.input :slack_username %>
+  <%= form.input :slack_username,
+                 required: true,
+                 pattern: '^[^@]\w+[^ ]$',
+                 input_html: {
+                   title: I18n.t('employees.errors.slack_username')
```

Comment:
Sorry it has taken me so long to get back to this! I like that a lot. I uncertain why the check on line 18 is needed. The only way we could get to that check is after saying: The record is valid, this username exists in your slack organization, the record failed to save, oh look, this username exists in your slack organization; which is exactly what we want to happen in the second branch of the conditional. I don't understand why what is considered valid in one context is suddenly an error further down in the conditional tree. 

Perhaps I am missing something, but it seems like the inclusion validation combined with the uniqueness validation on `Employee#slack_username` should be sufficient here. Am I missing a case?